cmake/modules/dts.cmake

```diff
@@ -158,15 +161,10 @@ if(DTC_OVERLAY_FILE)
     )
 endif()
 
-set(i 0)
 foreach(dts_file ${dts_files})
-  if(i EQUAL 0)
-    message(STATUS "Found BOARD.dts: ${dts_file}")
-  else()
+  if(dts_file MATCHES ".*[.]overlay$")
```

Comment:
this actually changes printing behavior.

Existing on `zephyr/main`, if you specify `-DDTS_SOURCE=<path>/foo.dts`, the following is printed:
```
-- Found BOARD.dts: /<path>/foo.dts
```

but with those changes, nothing is printed when specifying: `-DDTS_SOURCE=` because this line only prints for files ending with `.overlay`.
and board file is now printed here when `DTS_SOURCE` is not given by caller:
https://github.com/zephyrproject-rtos/zephyr/blob/668ded1f55b2cd7cd4a965daeb3b63a042f0f746/cmake/modules/dts.cmake#L131

especially is system-dt and sysbuild is expected in future to generate and use `DTS_SOURCE` I believe that it is important that the list of `dts_files` being used are always printed, as they are today.
Also for the case when the files specified by the caller is not ending in `.overlay` cause the files are still used by the build system.